src/transcript_tool.py

```diff
@@ -56,39 +88,153 @@ def main(in_path: str, out_path: str, model_path: str) -> None:
 
     # fetch model
     model = SpeechRecognitionModel(model_path)
-    print(f"\nModel: {model_path} fetched\n")
+    logging.info("Model loaded from: %s", model_path)
+
+    # create interim folders for processing slices & segments
+    if not os.path.isdir("../data/interim/slices"):
+        os.makedirs(os.path.abspath("../data/interim/slices"))
+    if not os.path.isdir("../data/interim/segments"):
+        os.makedirs(os.path.abspath("../data/interim/segments"))
 
     # get file_list
-    print(f"Creating transcript for files in {in_path} \nOutput dir: {out_path}")
+    logging.info("Loading input .mp3 files from: %s", in_path)
     full_paths = [
         os.path.abspath(os.path.join(in_path, f)) for f in os.listdir(in_path)
     ]
     mp3s = [e for e in full_paths if os.path.isfile(e) and e[-4:] == ".mp3"]
 
     for mp3 in mp3s:
+        # check outpout to avoid reruns
+        if is_processed(mp3, out_path):
+            logging.info("%s already has a transcirpt in %s", mp3, out_path)
+            continue
         # create transcript
         make_transcript(mp3, out_path, model)
 
 
 def make_transcript(
     audio_file: str, out_path: str, model: SpeechRecognitionModel
-) -> None:  # what is the model type?
```

Comment:
use type inference to determine the type of any unknown type:
https://mypy.readthedocs.io/en/stable/type_inference_and_annotations.html#
reveal_type(X) will run type inference on X and prints out its type
NB: reveal_type() will only run during type checking (mypy) and should be removed when you wanna run the code

Reply:
noted!